File: src/main/java/br/com/blz/testjava/domain/repository/ProductRepository.java
package br.com.blz.testjava.domain.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import br.com.blz.testjava.domain.model.Product;

@Repository
public interface ProductRepository extends JpaRepository<Product, Long> {

	Optional<Product> findBySku(Long sku);
}

File: src/main/java/br/com/blz/testjava/domain/repository/ProductRepository.kt
package br.com.blz.testjava.domain.repository

import br.com.blz.testjava.domain.entity.Product

interface ProductRepository {
  fun save(product: Product): Product

  fun findById(sku: String): Product?

  fun existsById(sku: Int): Boolean

  fun deleteById(sku: Int)
}
